components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/WebsocketInboundHandler.java

```diff
@@ -86,6 +86,14 @@ private void initializeDataPublisher() {
     public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
 
         String channelId = ctx.channel().id().asLongText();
+
+        // This block is for the health check of the ports 8099 and 9099
+        if (msg instanceof FullHttpRequest && !((FullHttpRequest) msg).headers().contains(HttpHeaders.UPGRADE)
```

Comment:
shall we add an additional AND condition to verify if the headers are not null?

Reply:
added.